xonsh/proc.py

```diff
@@ -552,8 +550,6 @@ def __bool__(self):
     def __eq__(self, other):
         return self.returncode == other
 
-    def __hash__(self):
-        return hash(self.returncode)
```

Comment:
I'm not sure I agree, but that's fine.  I thought it was nice to be able just to compare the object directly against the numbers.  But typing `.rtn` isn't too much extra hassle.  If it ends up being too painful, we can always add these things back in later.

In that case, let's add `rtn` as a short-hand for `returncode`.

And if we're getting rid of `__hash__`, I think we should also get rid of `__eq__` (and update the documentation appropriately) for the sake of consistency.